### ISSUE TITLE
sklearn f1_score method has 'binary' as average default

### ISSUE THREAD

**Issue by gianpDomiziani** [microsoft/FLAML#75]
sklearn f1_score makes use of 'binary' as average default parameter. It means, if the problem is a multiclass ones, it must be changed the average parameter to ones of ['micro', 'macro', 'weighted', 'samples']. In the ml module the sklearn_metric_loss_score is called without specifiyng that parameter. Consequently, at the moment, if a multiclass problem and f1 metric are choosen, a problem arise.

One solution, could be to set average='samples' when the task is multiclass:softmax. 
However, choosing one of the above list options, depends on the nature of the labels (balanced/unbalanced). It may be interesting to automate the process of choosing the best metric looking to the nature of the labels.

Any idea?
![err_flaml](https://user-images.githubusercontent.com/50718324/115834666-c1f10a00-a415-11eb-83cf-752072ce75b5.png)

settings = {
    "time_budget": TIME_BUDGET, 
    "metric": 'f1', 
    "estimator_list": ['lgbm'], 
    "task": 'classification',   
    "log_file_name": 'flaml_lgb.log',  
}

**Comment by qingyun-wu** [comment]
Hi @gianpDomiziani,

Thanks for reporting this issue. It seems f1 score is not a commonly used metric for multiclass classification tasks. So it is not clear to me how we should properly chose the 'best' metric in this case.

Among the several average options ['micro', 'macro', 'weighted', 'samples'], 'micro', 'macro', 'weighted' all work well. Do you think it is reasonable to just set the average option to be one of these three in the case of the multiclass classification task?

Thank you!

**Comment by sonichi** [comment]
or maybe we should ask users to specify 'micro_f1', 'macro_f1', etc. as metrics.

**Comment by gianpDomiziani** [comment]
Hi @qingyun-wu, yes, you right, it is not very common to use f1 score for multiclass tasks. However, sometimes looking to the f1-micro / f1-macro could be required. By very good, I mean that if the dataset is unbalanced a micro f1 should be used more tough than a macro f1, however it could be an unnecessary complication
Instead, I think the solution proposed by @sonichi should be great. If the task is multiclass f1 shouldn't be a valid option, while micro_f1/macro_f1 yes.

**Comment by sonichi** [comment]
@qingyun-wu do you agree? If so, I'd encourage @gianpDomiziani to create a PR. Thanks.

**Comment by qingyun-wu** [comment]
I agree. Thanks!
